packages/ember-data/lib/system/model/states.js

```diff
@@ -343,7 +345,7 @@ var DirtyState = {
 
     rolledBack: function(record) {
       get(record, 'errors').clear();
-      record.triggerLater('ready');
+      record.transitionTo('loaded.saved');
```

Comment:
I agree that we need add test case for rollbacking new uncommitted invalid record.
For me it make sense to handle it same as rollbacking uncommitted valid record.